src/tools/sc-hsm-tool.c

```diff
@@ -803,10 +821,9 @@ static void ask_for_password(char **pwd, int *pwdlen)
 
 
 
-static int generate_pwd_shares(sc_card_t *card, char **pwd, int *pwdlen, int password_shares_threshold, unsigned int password_shares_total)
+static int generate_pwd_shares(sc_card_t *card, char **pwd, int *pwdlen, int password_shares_threshold, int password_shares_total)
```

Comment:
That's OK. I'm fine with `-1` as default value. I am talking about the following different layers:
1. `create_dkek_share` which creates shares from a given user input (e.g. negative values)
2. `generate_pwd_shares` which creates shares from well defined `password_shares_threshold` and `password_shares_total` (e.g. at least positive/unsigned values)

There is no need for a rewrite when keeping these logical layers alive. But, what ever, it's way too unimportant to spend so much time on this discussions ;-)